dlt/pipeline/__init__.py

```diff
@@ -193,6 +196,8 @@ def attach(
     destination: TDestinationReferenceArg = None,
     staging: TDestinationReferenceArg = None,
     progress: TCollectorArg = _NULL_COLLECTOR,
+    dataset_name: str = None,
+    sync_if_missing: bool = False,
     **injection_kwargs: Any,
 ) -> Pipeline:
```

Comment:
I'm not sure about the changes to `def attach`. Maybe we should have something like `attach_remote` with a reduced arg set that does this and leave the attach unchanged.